smartsim/settings/launch_settings.py

```diff
@@ -165,6 +174,15 @@ def env_vars(self, value: dict[str, str | None]) -> None:
 
         :param value: The new environment mapping
         """
+        if not (
+            isinstance(value, t.Mapping)
+            and all(
+                isinstance(key, str) and isinstance(val, str)
+                for key, val in value.items()
```

Comment:
Similar discrepancy here: type hint asks for a `dict[str, str | None]`, but will error in this case
```py
launch_settings.env_vars = {"SOME_KEY": None}
```
Can we resolve?

Reply:
changed